src/NuGet.Clients/NuGet.CommandLine/Commands/TrustedSignersCommand.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Packaging.Signing;
+using static NuGet.Commands.TrustedSignersArgs;
+
+namespace NuGet.CommandLine.Commands
+{
+    [Command(typeof(NuGetCommand), "trusted-signers", "TrustedSignersCommandDescription",
+        MinArgs = 0,
+        MaxArgs = 2,
+        UsageSummaryResourceName = "TrustedSignersCommandUsageSummary",
+        UsageExampleResourceName = "TrustedSignersCommandUsageExamples")]
+    public class TrustedSignersCommand : Command
+    {
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandNameDescription")]
+        public string Name { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandServiceIndexDescription")]
+        public string ServiceIndex { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandCertificateFingerprintDescription")]
+        public string CertificateFingerprint { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandFingerprintAlgorithmDescription")]
+        public string FingerprintAlgorithm { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandAllowUntrustedRootDescription")]
+        public bool AllowUntrustedRoot { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandAuthorDescription")]
+        public bool Author { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandRepositoryDescription")]
+        public bool Repository { get; set; }
+
+        [Option(typeof(NuGetCommand), "TrustedSignersCommandOwnersDescription")]
+        public ICollection<string> Owners { get; set; }
+
+        internal ITrustedSignersCommandRunner TrustedSignersCommandRunner { get; set; }
+
+        internal TrustedSignersCommand() : base()
+        {
+            Owners = new List<string>();
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            var actionString = Arguments.FirstOrDefault();
+
+            TrustedSignersAction action;
+            if (string.IsNullOrEmpty(actionString))
+            {
+                action = TrustedSignersAction.List;
+            }
+            else if (!Enum.TryParse(actionString, ignoreCase: true, result: out action))
+            {
+                throw new CommandLineArgumentCombinationException(string.Format(CultureInfo.CurrentCulture, NuGetResources.Error_UnknownAction, actionString));
```

Comment:
aren't we already handing it across all the command line options?

Reply:
We are only handling it whenever we have options that we don't understand or if you have more or less arguments than the ones set in the `MaxArguments` or `MinArguments` properties of the command. However, this validation inside the command itself, when it got some valid arguments and after validating them it decided that the combination is not supported.

This is mostly useful for commands that have multiple sub commands like `verify` and `trusted-signers`

Reply:
I don't think you need the new exception type. 
https://github.com/NuGet/NuGet.Client/blob/7ca842c4386e287ba456a115979a7946766509df/src/NuGet.Clients/NuGet.CommandLine/Commands/SignCommand.cs#L116

Reply:
I could do the same thing as `SignCommand` and just throw `ArgumentException`, that results on just the error showing in the command line. I did this to have a similar behavior to when invalid arguments are passed so that you could see the error and then help being displayed. I though that was a more consistent behavior.

Nevertheless, we don't have a lot of consistency between how commands work in nuget.exe so I don't how any strong opinions about the path this should take.